src/compile.ts

```diff
@@ -336,47 +366,49 @@ export function compile(
          */
         scope?: ts.Node
       ): ts.Expression {
-        if (
-          !ts.isFunctionDeclaration(impl) &&
-          !ts.isArrowFunction(impl) &&
-          !ts.isFunctionExpression(impl) &&
-          !ts.isConstructorDeclaration(impl) &&
-          !ts.isMethodDeclaration(impl)
-        ) {
-          throw new Error(
-            `Functionless reflection only supports function parameters with bodies, no signature only declarations or references. Found ${impl.getText()}.`
-          );
-        }
+        return errorBoundary(() => {
+          if (
+            !ts.isFunctionDeclaration(impl) &&
+            !ts.isArrowFunction(impl) &&
+            !ts.isFunctionExpression(impl) &&
+            !ts.isConstructorDeclaration(impl) &&
+            !ts.isMethodDeclaration(impl)
+          ) {
+            throw new Error(
+              `Functionless reflection only supports function parameters with bodies, no signature only declarations or references. Found ${impl.getText()}.`
+            );
+          }
 
-        if (impl.body === undefined) {
-          throw new Error(
-            `cannot parse declaration-only function: ${impl.getText()}`
-          );
-        }
-        const body = ts.isBlock(impl.body)
-          ? toExpr(impl.body, scope ?? impl)
-          : newExpr(NodeKind.BlockStmt, [
-              ts.factory.createArrayLiteralExpression([
-                newExpr(NodeKind.ReturnStmt, [
-                  toExpr(impl.body, scope ?? impl),
+          if (impl.body === undefined) {
+            throw new Error(
+              `cannot parse declaration-only function: ${impl.getText()}`
+            );
```

Comment:
Synth Error?

Reply:
I don't expect the user to do anything for this case - it's internal.

Reply:
You did remind me to add a test-case though

Reply:
It only happens when we hoist registrations for function declarations in a block and that code just ignores any declaration-only functions - this is a bit like assertNever in that it catches a case that should never happen.